tools/accuracy_checker/accuracy_checker/preprocessor/brats_preprocessing.py

```diff
@@ -0,0 +1,176 @@
+"""
+Copyright (c) 2019 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import numpy as np
+from scipy.ndimage import interpolation
+
+from ..config import ConfigError, BaseField, NumberField, ListField, StringField
+from ..preprocessor import Preprocessor
+from ..utils import get_or_parse_value
+
+
+class Resize3D(Preprocessor):
+    __provider__ = 'resize3d'
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'size': BaseField(optional=True, description='Specifies resize'),
+        })
+        return parameters
+
+    def configure(self):
+        self.shape = get_or_parse_value(self.config.get('size'), (128, 128, 128), casting_type=int)
+        self.shape = (4,) + self.shape
+
+    def process(self, image, annotation_meta=None):
+        data = np.asarray(image.data)
+        if len(data.shape) != len(self.shape):
+            raise RuntimeError('Shape of original data and resize shape are mismatched for {} preprocessor '
+                               '(data shape - {}, resize shape - {})'.format(self.__provider__, data.shape, self.shape))
+
+        factor = [float(o) / i for i, o in zip(data.shape, self.shape)]
+        image.data = interpolation.zoom(data, zoom=factor, order=1)
+
+        return image
+
+
+class CropBraTS(Preprocessor):
+    __provider__ = 'crop_brats'
+
+    def process(self, image, annotation_meta=None):
+        def bbox3(img):
+            rows = np.where(np.any(np.any(img, axis=1), axis=1))
```

Comment:
the construction looks very strange, can we simplify it?

Reply:
It can be simplified, but code would be expanded :)